portfolio/src/main/webapp/script.js

```diff
@@ -269,3 +269,38 @@ function buildInfoWindowInput(lat, lng) {
 
   return containerDiv;
 }
+
+async function getLoginStatusUsingAsyncAwait() {
+
+  var section = document.getElementById('login-container');
+
+  fetch('/login').then(response => response.json()).then((status) => {
+    // status is an object, not a string, so we have to
+    // reference its fields to create HTML content
+
+    const statusListElement = document.getElementById('login-container');
+    statusListElement.innerHTML = '';
+    // If logged in show the container.
+    if (status.userEmail) {
+      section.className += " w3-show";
+      statusListElement.appendChild(
+          createListElement('Hello ' + status.userEmail +'!'));
+      statusListElement.appendChild(
+          createListElement('Logout: ' + status.logoutUrl));
+    }
+    // If not logged in display link to login page.
+    else {
+      statusListElement.appendChild(
+        createListElement('Hello stranger.'));
+      section.className = section.className.replace(" w3-show", "");
+      document.getElementById('login-link-container').className += " w3-show";
```

Comment:
Should there be a similar-but-opposite line in the "if" branch here, that e.g. hides/removes the w3-show class from login-link-container? (Also, see my previous question - is this supposed to be the same ID as either "section" or "statusListElement" - either maybe use those or update the ID?)